helmfile/charts/networkpolicy-workload/templates/ingress-nginx/controller.yaml

```diff
@@ -0,0 +1,98 @@
+{{ if .Values.ingressNginx.enabled }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: controller
+  namespace: ingress-nginx
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/component: controller
+  policyTypes:
+    - Ingress
+    - Egress
+  ingress:
+    - from:
+      {{- if .Values.ingressNginx.ingressOverride.enabled }}
+        {{- range $IP := .Values.ingressNginx.ingressOverride.ips }}
+        - ipBlock:
+            cidr: {{ $IP }}
+        {{- end }}
+      {{- else }}
+        {{- range $IP := .Values.global.wcIngress.ips }}
+        - ipBlock:
+            cidr: {{ $IP }}
+        {{- end }}
+      {{- end }}
+      ports:
+        - protocol: TCP
+          port: 443
+        - protocol: TCP
+          port: 80
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: monitoring
+          podSelector:
+            matchLabels:
+              app.kubernetes.io/instance: kube-prometheus-stack-prometheus
+      ports:
+        - port: 10254
+    - from:
+        - namespaceSelector:
+            matchLabels:
+              kubernetes.io/metadata.name: monitoring
+        - podSelector:
+            matchLabels:
+              app.kubernetes.io/instance: kube-prometheus-stack
+      ports:
+        - port: 10250
```

Comment:
What is this rule for? It feels odd that it is allowing the whole kube-prometheus stack (I think several apps have this label). Could you use a more specific label?

Reply:
I think, this is some sort of leftover.  As, I double checked, ingress-nginx pod does not listen on 10250 at all. 

hence, removing it.